test/functional/apps/discover/group3/_time_field_column.ts

```diff
@@ -0,0 +1,455 @@
+
+      await PageObjects.unifiedFieldList.clickFieldListItemAdd('@timestamp');
+      await retry.try(async () => {
+        expect(await dataGrid.getHeaderFields()).to.eql(['bytes', 'extension', '@timestamp']);
+      });
+
+      await PageObjects.discover.saveSearch(
+        `${SEARCH_WITH_SELECTED_COLUMNS_AND_TIMESTAMP}${savedSearchSuffix}`,
+        true
+      );
+      await PageObjects.discover.waitUntilSearchingHasFinished();
+
+      await dataGrid.clickMoveColumnLeft('@timestamp');
+      await retry.try(async () => {
+        expect(await dataGrid.getHeaderFields()).to.eql(['bytes', '@timestamp', 'extension']);
+      });
+
+      await dataGrid.clickMoveColumnLeft('@timestamp');
+      await retry.try(async () => {
+        expect(await dataGrid.getHeaderFields()).to.eql(
+          // FIXME as a part of https://github.com/elastic/kibana/issues/174074
+          isTextBased && !hideTimeFieldColumnSetting
+            ? ['bytes', 'extension']
+            : ['@timestamp', 'bytes', 'extension']
+        );
+      });
+
+      await PageObjects.unifiedFieldList.clickFieldListItemRemove('@timestamp');
+      await retry.try(async () => {
+        expect(await dataGrid.getHeaderFields()).to.eql(
+          hideTimeFieldColumnSetting || !hasTimeField || isTextBased
+            ? ['bytes', 'extension']
+            : ['@timestamp', 'bytes', 'extension']
+        );
+      });
+
+      // check in Dashboard
+      await PageObjects.common.navigateToApp('dashboard');
+      await PageObjects.dashboard.clickNewDashboard();
+      await dashboardAddPanel.clickOpenAddPanel();
+      await dashboardAddPanel.addSavedSearch(`${SEARCH_WITH_SELECTED_COLUMNS}${savedSearchSuffix}`);
+      await dashboardAddPanel.closeAddPanel();
+      await PageObjects.header.waitUntilLoadingHasFinished();
+
+      await retry.try(async () => {
+        expect(await dataGrid.getHeaderFields()).to.eql(
+          hideTimeFieldColumnSetting || !hasTimeField || isTextBased
+            ? ['bytes', 'extension']
+            : ['@timestamp', 'bytes', 'extension']
+        );
+      });
+
+      await dashboardPanelActions.removePanelByTitle(
+        `${SEARCH_WITH_SELECTED_COLUMNS}${savedSearchSuffix}`
+      );
+      await PageObjects.header.waitUntilLoadingHasFinished();
+      await dashboardAddPanel.clickOpenAddPanel();
+      await dashboardAddPanel.addSavedSearch(
+        `${SEARCH_WITH_SELECTED_COLUMNS_AND_TIMESTAMP}${savedSearchSuffix}`
+      );
+      await dashboardAddPanel.closeAddPanel();
+      await PageObjects.header.waitUntilLoadingHasFinished();
+
+      await retry.try(async () => {
+        expect(await dataGrid.getHeaderFields()).to.eql(['bytes', 'extension', '@timestamp']);
+      });
+    }
+
+    // Add `true` to the array to test locally with non default `doc_table:hideTimeColumn` setting too. I would not recommend committing with `true` as it doubles the whole test suit time.
```

Comment:
this is a good reminder that we should consider to deprecated this setting at a certain point, we could consider allowing the user remove the time column in a saved search. I don't see much sense, still having a global setting for this